### PR TITLE
Implement item light color providing

### PR DESCRIPTION
**Pull request by parzivail**
* Much like item light providing, this solution relies on an interface mixin that provides a default method.
* Color can be provided by an item each frame. I originally thought each tick to save performance, but well-optimized mods could take advantage of per-frame color.
* Implements uniforms for the main and offhand using a 3D float vector instead of a 4D one, since I don't know what transparency would mean in this context
* Will require shader support/implementation before the feature can be realized in clients

### CODE REVIEW

File: src/main/java/net/irisshaders/iris/api/v0/item/IrisItemColorProvider.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package net.irisshaders.iris.api.v0.item;
+
+import net.coderbot.iris.vendored.joml.Vector3f;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.item.ItemStack;
+
+public interface IrisItemColorProvider
```
Comment (maximumpower55) [inline src/main/java/net/irisshaders/iris/api/v0/item/IrisItemColorProvider.java:7]: This shouldn't be a separate interface Imo

File: src/main/java/net/irisshaders/iris/api/v0/item/IrisItemColorProvider.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+{
```
Comment (maximumpower55) [inline src/main/java/net/irisshaders/iris/api/v0/item/IrisItemColorProvider.java:8]: Why is the `{` on a separate line, this isn't csharp lmao

**Review by maximumpower55 — APPROVED** [review]
LGTM, just some minor nitpicks

File: src/main/java/net/irisshaders/iris/api/v0/item/IrisItemColorProvider.java (hunk 1 above)
Comment (parzivail) [reply src/main/java/net/irisshaders/iris/api/v0/item/IrisItemColorProvider.java:7]: Sure, I can merge it into `IrisItemLightProvider` if it's a better fit

File: src/main/java/net/irisshaders/iris/api/v0/item/IrisItemColorProvider.java (hunk 2 above)
Comment (parzivail) [reply src/main/java/net/irisshaders/iris/api/v0/item/IrisItemColorProvider.java:8]: oops, IDE settings

File: src/main/java/net/coderbot/iris/mixin/MixinItem.java [hunk 3]
```diff
@@ -1,9 +1,11 @@
...
 
+import net.irisshaders.iris.api.v0.item.IrisItemColorProvider;
 import net.irisshaders.iris.api.v0.item.IrisItemLightProvider;
 import net.minecraft.world.item.Item;
 import org.spongepowered.asm.mixin.Mixin;
 
 @Mixin(Item.class)
-public class MixinItem implements IrisItemLightProvider {
+public class MixinItem implements IrisItemLightProvider, IrisItemColorProvider
+{
```
Comment (maximumpower55) [inline src/main/java/net/coderbot/iris/mixin/MixinItem.java:10]: Same here why is the `{` on a separate line

File: src/main/java/net/irisshaders/iris/api/v0/item/IrisItemColorProvider.java (hunk 1 above)
Comment (IMS212) [reply src/main/java/net/irisshaders/iris/api/v0/item/IrisItemColorProvider.java:7]: Not sure if this should use JOML's Vector3f. This makes it hard for other mods to support without a second dependency, while MC's Vec3 would work fine.

**Pull request merged**